src/components/PopupWithForm.js

```diff
@@ -2,32 +2,36 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super(popupSelector);
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._inputList = this._popupForm.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
-    this._form = this._popup.querySelector(".modal__form");
-    this._inputList = this._form.querySelectorAll(".modal__input");
-    this._submitButton = this._form.querySelector(".modal__submit-button");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonValue = this._submitButton.textContent;
   }
 
   _getInputValues() {
-    const formValues = {};
+    const formInputs = {};
     this._inputList.forEach((input) => {
-      formValues[input.name] = input.value;
+      formInputs[input.name] = input.value;
     });
-    return formValues;
+    return formInputs;
   }
 
-  setInputValues(data) {
-    this._inputList.forEach((input) => {
-      input.value = data[input.name];
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`